sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HivePartitionFilteringSuite.scala

```diff
@@ -352,6 +352,26 @@ class HivePartitionFilteringSuite(version: String)
       dateStrValue)
   }
 
+  test("getPartitionsByFilter: ds<>20170101") {
+    testMetastorePartitionFiltering(
+      attr("ds") =!= 20170101,
```

Comment:
Could you add null and empty string as partition values into your test cases? It should include both data values and predicate values.

Reply:
For your reference, https://github.com/apache/spark/pull/30893 is the related PR